subsys/usb/device_next/class/usbd_uvc.c

```diff
@@ -1712,7 +1747,7 @@ static int uvc_init(struct usbd_class_data *const c_data)
 
 	atomic_set_bit(&data->state, UVC_STATE_INITIALIZED);
 
-	return 0;
+	return first_err;
```

Comment:
I do not entirely get the idea here. Is the user really care about what the first error was?

Reply:
The idea was to allow a few functions to fail due to being short on memory for the descriptors. In this case some errors would be ignored, with the error code stored for later. We then have the choice: return the first or last error code encountered.

But it is also possible to fail early, which would prevent the enumeration to happen when running out of configured memory. This would be a lot less confusing to debug and avoid such oddities.